### PR TITLE
Use Jetpack AI endpoint for product name suggestions

### LINKED ISSUE #38451: Use Jetpack AI endpoints for generating text
Modify the completion service used in https://github.com/woocommerce/woocommerce/pull/38392 to use the [Jetpack AI endpoint](https://github.com/Automattic/jetpack/blob/3be4a6e0afa8b267897dc18afd4e8163754df333/projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-ai.php#L52).

### PR DESCRIPTION
**Pull request by nima-karimi**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR replaces the OpenAI API with [Jetpack's completions API](https://github.com/Automattic/jetpack/blob/3be4a6e0afa8b267897dc18afd4e8163754df333/projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-ai.php#L52). Due to the limitations of Jetpack's API (lack of configurable options and chat history), I had to remove the temperature option and the retry functionality from the suggestions service. For some reason, the time to get a response slightly increased.

I added a check to the plugin's bootstrap to confirm that Jetpack is installed and active. If not, we will display an error message on the WP Admin. This has a side effect on the product description feature, too, but since we're [refactoring](https://github.com/woocommerce/woocommerce/pull/38407) it to also use Jetpack's API, it should be fine.

The completions service also checks that Jetpack is connected and returns an error if it is not.

**NOTE**: Currently, Jetpack caches the results of completion for 15 seconds so you might get duplicate results when clicking "Suggest Alternatives", but this will soon change (hence, the `skip_cache` option): https://github.com/Automattic/jetpack/pull/30938

Closes #38451

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

I used my local dev site to test this. But you can use an atomic site too.

1. Install the Jetpack plugin, activate and connect it to WPCOM
2. Add the sticker to your site (pe4Cmx-1cK-p2)
3. Navigate to the product edit page for an existing product that already has a title or description
4. Click on the Get Suggestions button and confirm you get the suggestion for your product name

### CODE REVIEW

**Comment by tommyshellberg** [comment]
I synced these changes to my Atomic site and it worked! 🥳 
Of note for anyone else testing: you don't need to set the `OPENAI_ENABLED` [blog sticker for Atomic sites](fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Sbcranv%2Sbcranv.cuc%3Se%3Qo2162po2%2338-og), only non WordPress.com/Atomic Sites.

**Comment by tommyshellberg** [comment]
> NOTE: Currently, Jetpack caches the results of completion for 15 seconds so you might get duplicate results when clicking "Suggest Alternatives", but this will soon change (hence, the skip_cache option)

Should we create a follow-up to send the `$skip_cache` param once the Jetpack plugin is updated for Atomic? It's been merged but not sure if it's been released yet.

**Comment by nima-karimi** [comment]
> I synced these changes to my Atomic site and all I had to do was set the OPEN_AI_KEY and it worked!

That's great. Do you mean that without the `OPEN_AI_KEY`, it didn't work? Because it doesn't have any dependencies on that. Or maybe it was caused by another place in the code? 🤔

**Comment by nima-karimi** [comment]
> Should we create a follow-up to send the $skip_cache param once the Jetpack plugin is updated for Atomic? It's been merged but not sure if it's been released yet.

I've added the `skip_cache` param here already: https://github.com/woocommerce/woocommerce/pull/38452/files#diff-5fe94266b47aa8d8a640dd2d4e6ca97a1cb4f3a00bf74865e80e5a07616ac576R60

So when they release that PR, we already use the feature.

File: plugins/woo-ai/includes/class-woo-ai-admin-notices.php [hunk 1]
```diff
@@ -18,4 +18,11 @@ class Woo_AI_Admin_Notices {
 	public function woocoommerce_not_installed() {
 		include_once dirname( __FILE__ ) . '/views/html-admin-missing-woocommerce.php';
```
Comment (joelclimbsthings) [inline plugins/woo-ai/includes/class-woo-ai-admin-notices.php:19]: I know this isn't from this PR, but  I think we want to update the associated view with this other notice. It refers to the Beta Tester plugin

**Review by joelclimbsthings — CHANGES_REQUESTED** [review]
Thanks for this @nima-karimi , looks to be working great! I left one minor comment, and I believe with this in place we can just remove the `text-completion` endpoint entirely?

File: plugins/woo-ai/includes/class-woo-ai-admin-notices.php (hunk 1 above)
Comment (nima-karimi) [reply plugins/woo-ai/includes/class-woo-ai-admin-notices.php:19]: Updated in https://github.com/woocommerce/woocommerce/pull/38452/commits/284c95c64ac2c99b12a64a994ba4c810997c867f

**Comment by nima-karimi** [comment]
@joelclimbsthings Thanks for the review. Yes, we can remove that one since we don't use it. I've removed it in https://github.com/woocommerce/woocommerce/pull/38452/commits/2cacbba82cfb173937f56a477fcf74d94f57828b

**Review by tommyshellberg — APPROVED** [review]
This is working well, I'm excited to see how it works when the completions endpoint accepts the `$skip_cache` argument. 🚢

**Pull request merged**